src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -3435,20 +3406,10 @@ extern "C" void * QCALLTYPE CallCatchFunclet(QCall::ObjectHandleOnStack exceptio
 
         EH_LOG((LL_INFO100, "Calling catch funclet at %p\n", pHandlerIP));
 
-#ifdef USE_FUNCLET_CALL_HELPER
-        // Invoke the catch funclet.
-        // Since the actual caller of the funclet is the assembly helper, pass the reference
-        // to the CallerStackFrame instance so that it can be updated.
-        CallerStackFrame* pCallerStackFrame = &exInfo->m_csfEHClause;
-        UINT_PTR *pFuncletCallerSP = &(pCallerStackFrame->SP);
-        dwResumePC = CallEHFunclet(OBJECTREFToObject(throwable),
-                                   CastHandlerFn(pfnHandler),
-                                   GetFirstNonVolatileRegisterAddress(pvRegDisplay->pCurrentContext),
-                                   pFuncletCallerSP);
+        dwResumePC = exInfo->m_frameIter.m_crawl.GetCodeManager()->CallFunclet(throwable, pHandlerIP, pvRegDisplay, exInfo, false /* isFilterFunclet */);
 
+#ifdef USE_FUNCLET_CALL_HELPER
         FixContext(pvRegDisplay->pCurrentContext);
```

Comment:
It would make sense to move the `FixContext` logic inside `EECodeManager::CallFunclet`. Afterall I don't expect it to be relevant for interpreter (ignoring the fact that we don't support interpreter on x86) and it only calls back into the code manager anyway.

Reply:
I've actually had it there, but moved it back. It is used for catch funclet only, so I would have to add another flag to the signature to distinguish catch. The context fixing is only relevant to the resuming after catch and that happens here, not in the code manager. I would actually prefer replacing the fixcontext here by its body and ifdef it on x86, since the function does nothing for all other architectures.